client/src/components/TextInput.jsx

```diff
@@ -11,9 +11,11 @@ const TextInput = ({
   maxLength,
   multiline,
   disabled,
+  isNumberVariant,
 }) => {
   const classes = `${label && "text-input__label"}
     ${subLabel ? "text-input__label--with-sublabel" : ""}`;
+  const inputType = isNumberVariant ? "number" : "text";
```

Comment:
I am not against this implementation. You could definitely still default to "text" if you used a variant prop that passed in the string of "number". But I think this is a good way to do it as well. If we have more variants we need to handle in future maybe then we would refactor this. But we will wait to do that if the need arises (and it may never need to change).